gtrending/cli.py

```diff
@@ -0,0 +1,187 @@
+import argparse
+import json
+
+from typing import Dict, Any
+
+from .fetch import (
+    fetch_repos,
+    fetch_developers,
+    languages_list,
+    spoken_languages_list,
+)
+
+
+def main():
+
+    parser = argparse.ArgumentParser('Gtrending')
+    subparser = parser.add_subparsers(dest='command')
+  
+    # Extract common arguments in parents
+    parent_json = argparse.ArgumentParser(add_help=False)
+    parent_json.add_argument(
+        '--json',
+        '-J',
+        action=argparse.BooleanOptionalAction,
+        default=False,
+    )
+
+    parent_filter = argparse.ArgumentParser(add_help=False)
+    parent_filter.add_argument(
+        '--language',
+        '--lang',
+        default=""
+    )
+    parent_filter.add_argument(
+        '--since',
+        type=str,
+        choices=['daily', 'monthly', 'weekly'],
+        default='daily'
+    )
```

Comment:
Let's have shorthands for these flags like `--json`. E.g., `--language`/`-l`, `--since`/`-s`, `--spoken-language`/`-S`. Maybe the sort flags don't need shorthands if you used s/S for since/spoken-language, up to you.